snowflake/src/test/java/org/apache/iceberg/snowflake/JdbcSnowflakeClientTest.java

```diff
@@ -47,6 +52,37 @@ public class JdbcSnowflakeClientTest {
   @Mock private JdbcSnowflakeClient.QueryHarness mockQueryHarness;
   @Mock private ResultSet mockResultSet;
 
+  @FunctionalInterface
+  private interface TestHelper {
+    void run();
+  }
+
+  private void generateSqlExceptionAndAssert(
+      TestHelper fn, Class expectedExceptionClass, String reason, int errorCode)
```

Comment:
Instead of passing reason and errorCode, could we directly pass the injectedException? This way we don't need to create different methods for each type of exception and errorcode that we inject.